src/bid_order.rs

```diff
@@ -227,23 +227,22 @@ pub fn migrate_bid_orders(
 
     require_version(">=0.16.2", &current_version)?;
 
-    // Migrate BidOrderV2 -> BidOrderV2
-    if VersionReq::parse(">=0.16.2, <0.18.2")?.matches(&current_version) {
-        // get all bid ids
-        let existing_bid_order_ids: Vec<Vec<u8>> = get_bid_storage_read::<BidOrderV2>(store)
+    // Migrate only BidOrderV2 -> BidOrderV3
+    if VersionReq::parse(">=0.16.2, <0.19.1")?.matches(&current_version) {
+        // get all bid ids for orders in a BidOrderV2 format
+        let existing_bid_order_v2_ids: Vec<Vec<u8>> = get_bid_storage_read::<BidOrderV2>(store)
             .range(None, None, Order::Ascending)
-            .map(|kv_bid| {
-                let (bid_key, _) = kv_bid.unwrap();
-                bid_key
-            })
+            // Get only the orders that are in a BidOrderV2 format
+            .filter_map(|kv_bid| kv_bid.ok().map(|record| record.0))
```

Comment:
This works when both BidOrderV2 and BidOrderV3 items exist (Only processes BidOrderV2 records)